extension-base/src/main/java/com/azure/autorest/extension/base/plugin/JavaSettings.java

```diff
@@ -124,7 +124,8 @@ public static JavaSettings getInstance()
                     host.getStringValue("customization-class"),
                     host.getBooleanValue("model-override-setter-from-superclass",
                             modelOverrideSetterFromSuperclassDefault),
-                    host.getBooleanValue("optional-constant-as-enum", false));
+                    host.getBooleanValue("optional-constant-as-enum", false),
+                    host.getBooleanValue("use-iterable", false));
```

Comment:
"use-iterable" maybe a bit too generic?

One question, does it affect client method parameter, or getter, or setter?

It seems we would prefer input as Iterable, but return value as Collection type.

Reply:
This will affect all. Making changes only to the client method and not the models can potentially cause some compilation errors. Also, we need to identify input-only models, output-only models and input-output models in autorest java. .NET already does this and it would be a good addition to Java as well. When we have this feature, we can consider generating List/Iterable depending on how the models are used. For now, we'll change all List to Iterable when this flag is set and we can introduce more finer control later.